app/blogPosts.py

```diff
@@ -1,52 +1,87 @@
+# @login_required
+def create_post():
+    # handle POST method
+    if request.method == 'POST':
+        # process request data into convenient object vessel 
+        title = request.form.get('title')
+        # Logan Kiser: will require user/group at some point, hold off for
+        #              different branch/issue
+        # user = current_user.id
+        content = request.form.get('content')
+        tags = request.form.get('tags')
+        new_post = BlogPost(title=title, content=content, tags=tags)
+        # insert post into database via BlogPost object
+        db.session.add(new_post)
+        db.session.commit()
+        # render main blog page
+        return redirect(url_for("blogPosts.display_post", post_id=new_post.id))
+    # handle GET method
+    return render_template('blog/create.html', newPost=1, links=get_links())
+
+
+# Logan Kiser: Kabir had added some UPDATE & DELETE handlers. I tidied them
+#              up a tiny bit, but they aren't necessarily as robust as we
+#              would like them to be in production. This was outside the scope
+#              of the GH issue.
+@blogPosts.route("/blog/<int:post_id>/update", methods=['GET', 'POST'])
+# Logan Kiser: hang tight on requiring credentials for flask-principal branch
+# @login_required
 def update_post(post_id):
-    post = Post.query.get_or_404(post_id)
-    if post.author != current_user:
-        abort(403)
-    form = postForm()
-    if form.validate_on_submit():
-        post.title = form.title.data
-        post.content = form.content.data
+    post = BlogPost.query.get_or_404(post_id)
+    if request.method == 'POST':
+        # first retrieve object from DB
+        updated_post = BlogPost.query.get(post_id)
+        # update appropriate fields
+        # this assumes all fields are passed via request
+        updated_post.title = request.form.get('title')
+        updated_post.content = request.form.get('content')
+        updated_post.tags = request.form.get('tags')
         db.session.commit()
         flash('Your post has been updated!', 'success')
-        return redirect(url_for('post', post_id=post.id))
-    elif request.method == 'GET':
-        form.title.data = post.title
-        form.content.data = post.content
-    return render_template('create_post.html', title='Update Post',
-                           form=form, legend='Update Post')
+        return redirect(url_for('blogPosts.display_post', post_id=post.id))
+    # populate form with existing data and allow user to alter blog post
+    return render_template('blog/update.html', post=post, links=get_links())
 
 
-@blogPosts.route("/post/<int:post_id>/delete", methods=['POST'])
-@login_required
+@blogPosts.route("/blog/<int:post_id>/delete", methods=['GET'])
+# Logan Kiser: hang tight on requiring credentials for flask-principal branch
+# @login_required
 def delete_post(post_id):
-    post = Post.query.get_or_404(post_id)
-    if post.author != current_user:
-        abort(403)
```

Comment:
Plan to add similar checking in the flask-principal branch.

Reply:
Yes.  I would prefer a group membership model.